File: 0646-maximum-length-of-pair-chain/0646-maximum-length-of-pair-chain.cpp
class Solution {
public:
    static bool compare(vector<int> &a,vector<int> &b)
    {
        return a[1]<b[1];
    }
    
    
    
    int findLongestChain(vector<vector<int>>& pairs) {
        sort(pairs.begin(),pairs.end(),compare);
       
        int count=1;
        int current_end=pairs[0][1];
        for(int i=1;i<pairs.size();i++)
        {
            if(pairs[i][0]>current_end)
                {current_end=pairs[i][1];
                count++;}
        }
        return count;
    }
};

File: 0646-maximum-length-of-pair-chain/0646-maximum-length-of-pair-chain.py
class Solution:
    def findLongestChain(self, pairs: List[List[int]]) -> int:
        if len(pairs)==0:
            return 0
        li=sorted(pairs,key=lambda x:x[1])
        cnt=1
        end=li[0][1]
        
        for i in li:
            if i[0]>end:
                cnt+=1
                end=i[1]
                
        return cnt
        
        